### PR TITLE
Automatically load tags from the filesystem

### PR DESCRIPTION
**Pull request by joshuaclayton**

### CODE REVIEW

File: src/Unused/TagsSource.hs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+loadTagsFromFile :: IO [String]
+loadTagsFromFile = fmap tokensFromTags tagsContent
+
+tokensFromTags :: String -> [String]
+tokensFromTags =
+    filter tagRemovalRegex . nub . map firstItemPerList . linesToTokensList
+  where
+    linesToTokensList = map (T.splitOn "\t" . T.pack) . lines
+    firstItemPerList = T.unpack . head
```
Comment (jsteiner) [inline src/Unused/TagsSource.hs]: maybe `firstToken`?

File: src/Unused/TagsSource.hs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+tagRemovalRegex :: String -> Bool
+tagRemovalRegex = not . matchRegex "^!_TAG"
+
+tagsContent :: IO String
+tagsContent = findFile possibleTagsFileDirectories "tags" >>= maybe (return "") readFile
```
Comment (jsteiner) [inline src/Unused/TagsSource.hs]: Should this raise an error if it can't read the file?

Comment (joshuaclayton) [inline src/Unused/TagsSource.hs]: Yeah, good call. I captured this with an `Either` in this commit: 83b3be098609cf23363ab67943ca787fad55c1e2

**Comment by joshuaclayton** [comment]
This will also resolve #35

**Pull request merged**